server/pulp/server/db/model/dispatch.py

```diff
@@ -614,7 +616,9 @@ def save(self, fields_to_set_on_insert=None):
         update = {'$set': stuff_to_update}
         if set_on_insert:
             update['$setOnInsert'] = set_on_insert
-        self.get_collection().update(
-            {'task_id': task_id},
-            update,
-            upsert=True)
+        if mongoengine.get_version()  < "0.9.0":
+            self.get_collection().update({'task_id': task_id}, update, upsert=True,
+                                         write_option={'w':1, 'fsync':True})
+        else:
+            self.get_collection().update({'task_id': task_id}, update, upsert=True,
+                                         write_concern={'w':1, 'fsync':True})
```

Comment:
Ugh, I can't remember if you can do thing = x if y else z in Python 2.6, so I guess we should make sure that's allowed, but the rest of the above pattern I think is still a little cleaner to avoid the repetition, so I think we should go with something like it at least.